benchexec/tablegenerator/react-table/src/components/SelectColumn.js

```diff
@@ -131,74 +175,103 @@ export default class SelectColumn extends React.Component {
   };
 
   // -------------------------Handling-------------------------
-  handleSelecion = ({ target }) => {
-    const [tool, column] = target.name.split("--");
-    const value = target.type === "checkbox" ? target.checked : target.value;
-    const list = [...this.state.list];
+  // Toggles all columns of all runsets
+  toggleAllColsHidden = () => {
+    let hiddenCols = {};
+    this.props.tools.forEach((tool) => (hiddenCols[tool.toolIdx] = []));
 
-    list[tool].columns.find(
-      (el) => el.display_title === column,
-    ).isVisible = value;
+    if (this.state.isButtonOnDeselect) {
+      this.props.tools.forEach((tool) => {
+        hiddenCols[tool.toolIdx] = tool.columns.map((column) => column.colIdx);
+      });
+    }
 
-    this.checkTools(list);
+    this.setState((prevState) => ({
+      isButtonOnDeselect: !prevState.isButtonOnDeselect,
+      hiddenCols,
+    }));
   };
 
-  handleSelectColumns = ({ target }) => {
-    const value = target.type === "checkbox" ? target.checked : target.value;
-    const list = [...this.state.list];
+  // Toggles all columns of the runset with the id of the target
+  toggleToolHidden = ({ target }) => {
+    let toolIdx = parseInt(target.name.split("--")[1]);
```

Comment:
Wouldn't it be cleaner to add the index as parameter to `toggleToolHidden` and use a lambda that captures the value for the event handler instead of a direct method reference?

The less string parsing the better.

Reply:
Yes, absolutely. I picked up the parsing of the html tag's name from the existing code, but providing the data directly via lambdas is definitely cleaner. I've applied your suggestion here and at other similiar parts of the code.